Crash in AztecEditorFragment after getting bad remote URL

```
01-11 20:56:34.158 17678-17678/org.wordpress.android D/AndroidRuntime: Shutting down VM
01-11 20:56:34.160 17678-17678/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                       Process: org.wordpress.android, PID: 17678
                                                                       java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.posts.EditPostActivity}: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter value
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                           at android.os.Looper.loop(Looper.java:164)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6541)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                                        Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter value
                                                                           at org.wordpress.aztec.AztecAttributes.setValue(Unknown Source:9)
                                                                           at org.wordpress.android.editor.AztecEditorFragment.replaceMediaFileWithUrl(AztecEditorFragment.java:1838)
                                                                           at org.wordpress.android.ui.uploads.MediaUploadReadyProcessor.replaceMediaFileWithUrlInPost(MediaUploadReadyProcessor.java:23)
                                                                           at org.wordpress.android.ui.uploads.UploadService.updatePostWithMediaUrl(UploadService.java:544)
                                                                           at org.wordpress.android.ui.uploads.UploadService.updatePostWithCurrentlyCompletedUploads(UploadService.java:422)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.initializePostObject(EditPostActivity.java:431)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.onCreate(EditPostActivity.java:331)
                                                                           at android.app.Activity.performCreate(Activity.java:6975)
                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1213)
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892) 
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                                           at android.os.Looper.loop(Looper.java:164) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6541) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
01-11 20:56:34.165 919-7974/? W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.posts.EditPostActivity
01-11 20:56:34.166 2553-19372/? V/FA-SVC: Logging event: origin=auto,name=user_engagement(_e),params=Bundle[{firebase_event_origin(_o)=auto, engagement_time_msec(_et)=2382, firebase_screen_class(_sc)=PostsListActivity, firebase_screen_id(_si)=8653637150776347266}]
01-11 20:56:34.168 919-7974/? W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.posts.PostsListActivity
```
azte

Probably was because I actually gave a damn about this man, was hurting. I'm not a nasty, lying, cowardly, whore that can live with that sort of filth. Not the trash that you appear to be. You'll be meeting the true detective soon enough.                                                                                                                                                                        Sent via the BlackBerry Hub for Android                                                                                                                                                                           From: notifications@github.comSent: January 11, 2018 6:59 PMTo: WordPress-Android@noreply.github.comReply-to: reply@reply.github.comCc: subscribed@noreply.github.comSubject: [wordpress-mobile/WordPress-Android] Crash in AztecEditorFragment after getting bad remote URL (#7089)  01-11 20:56:34.158 17678-17678/org.wordpress.android D/AndroidRuntime: Shutting down VM
01-11 20:56:34.160 17678-17678/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                       Process: org.wordpress.android, PID: 17678
                                                                       java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.posts.EditPostActivity}: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter value
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                           at android.os.Looper.loop(Looper.java:164)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6541)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                                        Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter value
                                                                           at org.wordpress.aztec.AztecAttributes.setValue(Unknown Source:9)
                                                                           at org.wordpress.android.editor.AztecEditorFragment.replaceMediaFileWithUrl(AztecEditorFragment.java:1838)
                                                                           at org.wordpress.android.ui.uploads.MediaUploadReadyProcessor.replaceMediaFileWithUrlInPost(MediaUploadReadyProcessor.java:23)
                                                                           at org.wordpress.android.ui.uploads.UploadService.updatePostWithMediaUrl(UploadService.java:544)
                                                                           at org.wordpress.android.ui.uploads.UploadService.updatePostWithCurrentlyCompletedUploads(UploadService.java:422)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.initializePostObject(EditPostActivity.java:431)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.onCreate(EditPostActivity.java:331)
                                                                           at android.app.Activity.performCreate(Activity.java:6975)
                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1213)
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892) 
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                                           at android.os.Looper.loop(Looper.java:164) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6541) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767) 
01-11 20:56:34.165 919-7974/? W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.posts.EditPostActivity
01-11 20:56:34.166 2553-19372/? V/FA-SVC: Logging event: origin=auto,name=user_engagement(_e),params=Bundle[{firebase_event_origin(_o)=auto, engagement_time_msec(_et)=2382, firebase_screen_class(_sc)=PostsListActivity, firebase_screen_id(_si)=8653637150776347266}]
01-11 20:56:34.168 919-7974/? W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.posts.PostsListActivity

azte

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.

--- a/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java
+++ b/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java
@@ -1814,7 +1814,7 @@ public class AztecEditorFragment extends EditorFragmentAbstract implements
     public static String replaceMediaFileWithUrl(Context context, @NonNull String postContent,
                                                  String localMediaId, MediaFile mediaFile) {
         if (mediaFile != null) {
-            String remoteUrl = Utils.escapeQuotes(mediaFile.getFileURL());
+            String remoteUrl = StringUtils.notNullStr(Utils.escapeQuotes(mediaFile.getFileURL()));
             // fill in Aztec with the post's content
             AztecParser parser = getAztecParserWithPlugins();
             SpannableStringBuilder builder = getCalypsoCompatibleStringBuilder(context, postContent, parser);

